src/html-conventions-lib/general-connectors-html-convention.xsl

```diff
@@ -688,4 +688,133 @@
         </xsl:if>
     </xsl:template>
 
+    <xd:doc>
+        <xd:desc>[common-name-unique] - The connector role name $value$ is not unique. </xd:desc>
+        <xd:param name="connector"/>
+    </xd:doc>
+    <xsl:template name="connectorUniqueName">
+        <xsl:param name="connector"/>
+        <xsl:variable name="sourceRole" select="$connector/source/role/@name"/>
+        <xsl:variable name="targetRole" select="$connector/target/role/@name"/>
+
+        <xsl:if test="boolean($sourceRole)">
+            <xsl:variable name="elementsFound"
+                select="f:getElementByName($sourceRole, root($connector))"/>
+            <xsl:variable name="attributesFound"
+                select="f:getAttributeByName($sourceRole, root($connector))"/>
+
+            <xsl:sequence
+                select="
+                    if (count($elementsFound) > 1 or count($attributesFound) > 0) then
+                        f:generateHtmlError(fn:concat('The connector role name ', $sourceRole, ' is not unique.'))
+                    else
+                        ()
+                    
+                    "
+            />
+        </xsl:if>
+        <xsl:if test="boolean($targetRole)">
+            <xsl:variable name="elementsFound"
+                select="f:getElementByName($targetRole, root($connector))"/>
+            <xsl:variable name="attributesFound"
+                select="f:getAttributeByName($targetRole, root($connector))"/>
+
+            <xsl:sequence
+                select="
+                    if (count($elementsFound) > 1 or count($attributesFound) > 0) then
+                        f:generateHtmlError(fn:concat('The connector role name ', $targetRole, ' is not unique.'))
+                    else
+                        ()
+                    
+                    "
+            />
+        </xsl:if>
+    </xsl:template>
+
+
+    <xd:doc>
+        <xd:desc>[association-dependency-checker] - Connector role names are used in other
+            connectors </xd:desc>
```

Comment:
"in other connector TYPES", this is what was meant, right?

Reply:
yes

Reply:
yes